### PR TITLE
Add session window sample

### PR DESCRIPTION
**Pull request by viliam-durina**

### CODE REVIEW

**Comment by viliam-durina** [comment]
This PR will compile, after hazelcast/hazelcast-jet#447 is merged.

**Comment by viliam-durina** [comment]
verify

File: streaming/session-windows/src/main/java/SessionWindowsSample.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import static com.hazelcast.jet.function.DistributedFunction.identity;
+import static com.hazelcast.jet.processor.DiagnosticProcessors.writeLogger;
+import static com.hazelcast.jet.processor.Processors.aggregateToSessionWindow;
+import static com.hazelcast.jet.processor.Processors.insertPunctuation;
+import static com.hazelcast.jet.samples.sessionwindows.ProductEventType.PURCHASE;
+import static com.hazelcast.jet.samples.sessionwindows.ProductEventType.VIEW_LISTING;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * A sample demonstrating accumulation into session windows. It processes
```
Comment (cangencer) [inline streaming/session-windows/src/main/java/SessionWindowsSample.java]: we should add a visual DAG

File: streaming/session-windows/src/main/java/SessionWindowsSample.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                toSet(e -> e.getProductEventType() == PURCHASE ? e.getProductId() : null)
+        );
+
+        // if you want to see events emitted from source, replace "GenerateEventsP::new"
+        // with "Processors.peekOutput(GenerateEventsP::new)"
+        Vertex source = dag.newVertex("source", GenerateEventsP::new)
+                           .localParallelism(1);
+        Vertex insertPunc = dag.newVertex("insertPunc", insertPunctuation(ProductEvent::getTimestamp,
+                () -> PunctuationPolicies.withFixedLag(100).throttleByMinStep(100)))
+                .localParallelism(1);
```
Comment (cangencer) [inline streaming/session-windows/src/main/java/SessionWindowsSample.java]: localParallelism should be default here.

File: streaming/session-windows/src/main/java/SessionWindowsSample.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        Vertex aggregateSessions = dag.newVertex("aggregateSessions",
+                aggregateToSessionWindow(5000, ProductEvent::getTimestamp, ProductEvent::getUserId, aggrOp));
```
Comment (cangencer) [inline streaming/session-windows/src/main/java/SessionWindowsSample.java]: might be better to extract this to a constant, such as SESSION_TIMEOUT

**Comment by viliam-durina** [comment]
verify

File: streaming/session-windows/src/main/java/SessionWindowsSample.java [hunk 4]
```diff
@@ -52,10 +52,36 @@
...
+ * The DAG is as follows:
+ * <pre>{@code
+ *          +--------------+
+ *          |    Source    |
+ *          +------+-------+
+ *                 |
+ *                 | ProductEvent
+ *                 |
+ *       +--------------------+
+ *       | Insert p|nctuation |
```
Comment (cangencer) [inline streaming/session-windows/src/main/java/SessionWindowsSample.java]: typo (p|nctuation)

**Pull request merged**